server/sync-server.ts

```diff
@@ -0,0 +1,92 @@
+import { Hono } from "hono"
+import { init as initQueue } from "@/queue"
+import config from "@/config"
+import { getLogger, LogMiddleware } from "@/logger"
+import { Subsystem } from "@/types"
+import { InitialisationError } from "@/errors"
+import metricRegister from "@/metrics/sharedRegistry"
+import {
+  isSlackEnabled,
+  startSocketMode,
+} from "@/integrations/slack/client"
+
+const Logger = getLogger(Subsystem.SyncServer)
+
+const app = new Hono()
+
+const honoMiddlewareLogger = LogMiddleware(Subsystem.SyncServer)
+
+// Add logging middleware
+app.use("*", honoMiddlewareLogger)
+
+// Health check endpoint
+app.get("/health", (c) => {
+  return c.json({ status: "ok", service: "sync-server" })
+})
+
+// Metrics endpoint for sync server
+app.get("/metrics", async (c) => {
+  try {
+    const metrics = await metricRegister.metrics()
+    return c.text(metrics, 200, {
+      "Content-Type": metricRegister.contentType,
+    })
+  } catch (err) {
+    Logger.error(err, "Error generating metrics")
+    return c.text("Error generating metrics", 500)
+  }
+})
+
+// Status endpoint to check sync operations
+app.get("/status", (c) => {
+  return c.json({
+    service: "sync-server",
+    status: "running",
```

Comment:
is there a way to validate if sync queue is running fine? we should be returning this only if the server is running fine